### PR TITLE
Add `ImageData` transformations to move between index and physical coordinates

### PR DESCRIPTION
**Pull request by user27182**
### Overview

Split from #6612

### CODE REVIEW

File: tests/core/test_grid.py [hunk 1]
```diff
@@ -1093,6 +1093,23 @@ def test_grid_points():
...
 
+def test_imagedata_index_to_physical_matrix():
+    # Create image with arbitrary translation (origin) and rotation (direction)
+    image = pv.ImageData()
+    translation = (1, 2, 3)
+    rotation = pv.Transform().rotate_vector((1, 2, 3), 30).matrix[:3, :3]
+    image.origin = translation
+    image.direction_matrix = rotation
+
+    # Do test
```
Comment (tkoyama010) [inline tests/core/test_grid.py:1104]:
```suggestion
```

File: pyvista/core/grid.py [hunk 2]
```diff
@@ -16,11 +16,13 @@
 if TYPE_CHECKING:
     from pyvista.core._typing_core import NumpyArray
 
+
```
Comment (tkoyama010) [inline pyvista/core/grid.py:19]:
```suggestion
```

File: tests/core/test_grid.py [hunk 3]
```diff
@@ -1093,6 +1093,23 @@ def test_grid_points():
     )
 
 
+def test_imagedata_index_to_physical_matrix():
+    # Create image with arbitrary translation (origin) and rotation (direction)
+    image = pv.ImageData()
+    translation = (1, 2, 3)
+    rotation = pv.Transform().rotate_vector((1, 2, 3), 30).matrix[:3, :3]
```
Comment (tkoyama010) [inline tests/core/test_grid.py:1100]:
```suggestion
    translation = (1, 2, 3)
    rotation = pv.Transform().rotate_vector(translation, 30).matrix[:3, :3]
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
Thank you. LGTM except for the commented part.

File: tests/core/test_grid.py [hunk 4]
```diff
@@ -1093,6 +1093,22 @@ def test_grid_points():
...
 
+def test_imagedata_index_to_physical_matrix():
+    # Create image with arbitrary translation (origin) and rotation (direction)
+    image = pv.ImageData()
+    translation = (1, 2, 3)
+    rotation = pv.Transform().rotate_vector(translation, 30).matrix[:3, :3]
+    image.origin = translation
+    image.direction_matrix = rotation
+
+    expected_transform = pv.Transform(rotation).translate(translation)
```
Comment (user27182) [inline tests/core/test_grid.py:1104]: better to use a more generic variable name (it's a bit confusing to have a var named `translate` inside of `rotate` method)
```suggestion
    vector = (1, 2, 3)
    rotation = pv.Transform().rotate_vector(vector, 30).matrix[:3, :3]
    image.origin = translation
    image.direction_matrix = rotation

    expected_transform = pv.Transform(rotation).translate(vector)
```

**Review by tkoyama010 — APPROVED** [review]

**Comment by user27182** [comment]
Needed to remove the rotation from the test. This is left over from the split from #6612 . We can add rotation to the test once `direction_matrix` is added.

**Review by tkoyama010 — APPROVED** [review]
Ah, I see, I forgot to check if it was removed from the test.

**Pull request merged**